### DISCUSSION TITLE
HDR Brightness mapping

### DISCUSSION THREAD

**Discussion by HernandoR** [MonitorControl/MonitorControl#1524] [Q&A]
Hi I've got an HDR Monitor, and I have got it working pretty good for SDR and HDR content with this app. However, when it comes to extremely bright content, i.e. over 400 nits, it will be cut off to blank. Any idea how I can set things up to make them mapping to an reasonable range?

**Comment by waydabber** [comment]
Hi there - what does it mean it will be cut off to blank? The entire screen will be black?

**Comment by HernandoR** [comment]
By blank I mean pure white, part of the image/video ( those brightest pixels will), for instance youtube<https://www.youtube.com/watch?v=z4swrq4ucfQ&t=556s> on 0:17,  the show ground will show full blank white on my external display (600 nit, full brightness, full contrast). More detail on my internal Display(1600 nit)

I can adjust the external display with (600 nit, full brightness, 50% contrast), then I can see the details, but anything else is gray, and not good.

The issue is out from a fresh fish: is there any way to suppress this margin cases? i.e. if I can set up certain mapping function such that brightness over 550nits can be mapped into 550-600 nits.

发件人: waydabber ***@***.***>
日期: 星期三, 2024年2月28日 15:46
收件人: MonitorControl/MonitorControl ***@***.***>
抄送: Zhen Liu ***@***.***>, Author ***@***.***>
主题: Re: [MonitorControl/MonitorControl] HDR Brightness mapping (Discussion #1524)

Hi there - what does it mean it will be cut off to blank? The entire screen will be black?

―

**Comment by waydabber** [comment]
Oh, you mean HDR clipping. Yes, this is a side-effect of gamma table manipulation (this works only on the SDR range and any brightness levels beyond this will be collapsed back to max SDR brightness).

I advise you to switch to [BetterDisplay](https://betterdisplay.pro):

- for the XDR screen you can use native XDR brightness upscaling (see under XDR Presets in the app menu) - this will expand the brightness range to 0-1600nits for the brightness slider and also preserve HDR colors.
- for an external display the app by default uses gamma table dimming so the HDR clipping issue will be present but you can switch to Metal based dimming/upscaling - this on Apple Silicon should work better on HDR content. Both will enable brightness upscaling for the HDR display.

**Reply by waydabber** [reply]
For more info on XDR/HDR brightness, methods, clipping, etc: https://github.com/waydabber/BetterDisplay/wiki/XDR-and-HDR-brightness-upscaling

**Comment by HernandoR** [comment]
I took a trial on Better Display, I’m not sure I have understanding the content on the wiki you suggested, but I’m afraid I haven’t make that works.

Moreover, I noticed this discussion: https://github.com/waydabber/BetterDisplay/discussions/2075, it seems that the HDR clipping is inevitable, I understand I should ask relating questions in BetterDisplay’s repo, but can you help me confirm if this discussion marked my issue is inevitable?

Regards

发件人: waydabber ***@***.***>
日期: 星期三, 2024年2月28日 17:08
收件人: MonitorControl/MonitorControl ***@***.***>
抄送: Zhen Liu ***@***.***>, Author ***@***.***>
主题: Re: [MonitorControl/MonitorControl] HDR Brightness mapping (Discussion #1524)

For more info on XDR/HDR brightness, methods, clipping, etc: https://github.com/waydabber/BetterDisplay/wiki/XDR-and-HDR-brightness-upscaling

―

**Reply by waydabber** [reply]
No, it is not inevitable. This happens only when the color table method is used for dimming or upscaling. With native XDR upscaling there is no such issue (but this only works on built-in XDR screens). With Metal upscaling (for that you need to turn off `Enable color table adjustments` under the display) clipping should not happen either (with some players on Apple Silicon - like Safari, QuickTime, TV - not affecting HDR content at all, and with others - like IINA - applying the brightening/dimming effect but without clipping).

**Reply by HernandoR** [reply]
You are Right! It’s much better now. One issue that occurs to me is the YouTube video will only show it correctly if my mouse in the video frame, i.e. the clipping remains if the mouse is in the comments section or something. But I believe that is a bug (or feature) of the browser.

Here‘s the settings for the reference to any followers
<img width="579" alt="image" src="https://github.com/MonitorControl/MonitorControl/assets/45709656/7d8053af-243d-4dce-ab94-451bbacd6157">
